src/main.rs

```diff
@@ -833,12 +857,46 @@ async fn run_with_tracing() -> Result<(), Error> {
             )?;
             let otel_sample_rate =
                 &std::env::var("OPENTELEMETRY_SAMPLE_RATE").unwrap_or_else(|_| "0.05".to_owned());
+
+            let otel_endpoint = if otel_endpoint.starts_with("https://") {
+                otel_endpoint
+            } else {
+                format!(
+                    "https://{}/",
+                    otel_endpoint.strip_suffix(":443").unwrap_or(&otel_endpoint)
+                )
+            };
```

Comment:
I think instead of munging the input, we can just take the endpoint as given here. When we update cloud to use the new version, we can update the input endpoint to point to https:// too.

Reply:
Also, does the endpoint format mean something to tonic? Does this use "HTTPS" instead of "GRPC" or something?

Reply:
Didn't know we could update the version and flags atomically!

It should be doing `grpc` over `http2` only: 
"A Rust implementation of [gRPC](https://grpc.io/), a high performance, open source, general RPC framework that puts mobile and HTTP/2 first.

[tonic](https://github.com/hyperium/tonic) is a gRPC over HTTP/2 implementation focused on high performance, interoperability, and flexibility."

unclear to me why the endpoint is so pedantic though, I think `from_shared` might be more conservative than the default connect's functions that use rustls?